paasta_tools/cli/cmds/status.py

```diff
@@ -347,81 +324,72 @@ def verify_instances(args_instances, service, clusters):
     return unverified_instances
 
 
-def paasta_args_mixer(args, service):
-    if args.deploy_group is None:
-        cluster_whitelist = args.clusters.split(",") if args.clusters is not None else []
-        if args.instances is not None:
-            instance_whitelist = verify_instances(args.instances, service, cluster_whitelist)
-        else:
-            instance_whitelist = []
+def get_filters(args):
+    if not any([args.clusters, args.instances, args.owner, args.deploy_group]):
+        service = figure_out_service_name(args, soa_dir=args.soa_dir)
     else:
-        clusters_instances = get_cluster_instance_map_for_service(
-            soa_dir=args.soa_dir,
-            service=service,
-            deploy_group=args.deploy_group,
-        )
-        cluster_whitelist = list(
-            (set(args.clusters.split(",")) & set(clusters_instances))
-            if args.clusters is not None else clusters_instances,
-        )
-
-        diff = set(cluster_whitelist) - set(clusters_instances)
-        if (args.clusters is not None and not cluster_whitelist) or diff:
-            paasta_print("The %s deploy_group doesn't have any instances of %s on %s." %
-                         (args.deploy_group, service, ', '.join(diff) or args.clusters))
-            return None
-
-        instances_set = set()
-        for cluster in cluster_whitelist:
-            for instance in clusters_instances[cluster].get('instances', []):
-                instances_set.add(instance)
-
-        instance_whitelist = list(
-            set(args.instances.split(",")) & instances_set
-            if args.instances is not None else instances_set,
-        )
-
-        diff = set(instance_whitelist) - instances_set
-        if (args.instances is not None and not instance_whitelist) or diff:
-            paasta_print("The %s deploy_group doesn't have any instances of %s matching %s." %
-                         (args.deploy_group, service, ', '.join(diff) or args.instances))
-            return None
-
-    PaastaArgs = namedtuple('PaastaArgs', ['cluster_whitelist', 'instance_whitelist'])
-    return PaastaArgs(
-        cluster_whitelist=cluster_whitelist,
-        instance_whitelist=instance_whitelist,
-    )
+        service = args.service
+
+    filters = []
+
+    if args.clusters:
+        filters.append(lambda conf: conf.get_cluster() in args.clusters.split(','))
+
+    if args.instances:
+        filters.append(lambda conf: conf.get_instance() in args.instances.split(','))
+
+    if args.service:
+        filters.append(lambda conf: conf.get_service() in service.split(','))
+
+    if args.deploy_group:
+        filters.append(lambda conf: conf.get_deploy_group() in args.deploy_group.split(','))
+
+    return filters
+
+
+def apply_args_filters(args):
```

Comment:
Could you add a docstring here?
We most certainly will reuse this function in other paasta subcommands, let's make it easier to reuse.